financialaid/views_test.py

```diff
@@ -548,6 +548,86 @@ def test_get_learner_price_for_enrolled_but_no_financial_aid_availability(self):
+        self.assert_http_status(self.client.put, self.skip_url, status.HTTP_200_OK)
+        assert FinancialAid.objects.count() == 3
+        financialaid = FinancialAid.objects.get(user=self.enrolled_profile3.user, tier_program__program=self.program)
+        assert financialaid.tier_program == self.tier_programs["100k"]
+        assert financialaid.status == FinancialAidStatus.SKIPPED
+        # Check logging
+        assert FinancialAidAudit.objects.count() == 1
+
+    def test_skipped_financialaid_object_updated(self):
+        """
+        Tests that an existing FinancialAid object is updated to have the status "skipped"
+        """
+        self.client.force_login(self.enrolled_profile2.user)
+        assert FinancialAidAudit.objects.count() == 0
+        # Check number of financial aid objects (two are created at test setup)
+        assert FinancialAid.objects.count() == 2
+        self.assert_http_status(self.client.put, self.skip_url, status.HTTP_200_OK)
+        assert FinancialAid.objects.count() == 2
+        self.financialaid_pending.refresh_from_db()
+        assert self.financialaid_pending.tier_program == self.tier_programs["100k"]
+        assert self.financialaid_pending.status == FinancialAidStatus.SKIPPED
+        # Check logging
+        assert FinancialAidAudit.objects.count() == 1
+
+    def test_financialaid_object_cannot_be_skipped_if_approved_or_rejected(self):
+        """
+        Tests that an existing FinancialAid object that has already been approved or rejected
+        cannot be skipped.
+        """
+        self.client.force_login(self.enrolled_profile2.user)
+        unpermitted_statuses = [
+            FinancialAidStatus.APPROVED,
+            FinancialAidStatus.AUTO_APPROVED,
+            FinancialAidStatus.REJECTED
+        ]
+        for unpermitted_status in unpermitted_statuses:
+            self.financialaid_pending.status = unpermitted_status
+            self.financialaid_pending.save()
+            self.assert_http_status(self.client.put, self.skip_url, status.HTTP_400_BAD_REQUEST)
+
+    def test_financialaid_object_cannot_be_skipped_if_aid_not_available(self):
+        """
+        Tests that a FinancialAid object cannot be skipped if program does not have financial
+        aid
+        """
+        self.client.force_login(self.enrolled_profile3.user)
+        self.program.financial_aid_availability = False
+        self.program.save()
+        self.assert_http_status(self.client.put, self.skip_url, status.HTTP_400_BAD_REQUEST)
+
+    def test_financialaid_object_cannot_be_skipped_if_not_enrolled_in_program(self):
+        """
+        Tests that a FinancialAid object cannot be skipped if the user is not enrolled in program
+        """
+        self.client.force_login(self.enrolled_profile3.user)
+        url = reverse("financial_aid_skip", kwargs={"program_id": self.program2.id})
+        self.assert_http_status(self.client.put, url, status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Can you also test requests for non existing programs and user not enrolled in the program